sparkctl/cmd/events.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2018 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"os"
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/util/interrupt"
+	crdclientset "k8s.io/spark-on-k8s-operator/pkg/client/clientset/versioned"
+	"strings"
+)
+
+var eventsCmd = &cobra.Command{
+	Use:   "events <name>",
+	Short: "Shows SparkApplication events",
+	Long:  `Shows events associated with SparkApplication of a given name`,
+	Run: func(cmd *cobra.Command, args []string) {
+		if len(args) != 1 {
+			fmt.Fprintln(os.Stderr, "must specify a SparkApplication name")
+			return
+		}
+
+		crdClientset, err := getSparkApplicationClient()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to get SparkApplication client: %v\n", err)
+			return
+		}
+
+		kubeClientset, err := getKubeClient()
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "failed to get KubeClient: %v\n", err)
+			return
+		}
+		if err := doShowEvents(args[0], crdClientset, kubeClientset); err != nil {
+			fmt.Fprintf(os.Stderr, "failed to check events of SparkApplication %s: %v\n", args[0], err)
+		}
+	},
+}
+
+func doShowEvents(name string, crdClientset crdclientset.Interface, kubeClientset kubernetes.Interface) error {
+	app, err := getSparkApplication(name, crdClientset)
+	if err != nil {
+		return fmt.Errorf("failed to get SparkApplication %s: %v", name, err)
+	}
+
+	// Since ResourceVersion of SparkApplication changes on CREATE/SUCCESS, we take all events for
```

Comment:
`ResourceVersion` changes whenever the object gets updated. Why this matters?

Reply:
The problem is that I cannot filter by "at least as early as resource version" since after application completes I will not see history for this application.